Automattic/jetpack
# Search: Migrate Classic Search, Instant Search, and Customberg to package

# Warning: #21690 must be resolved before merging this PR!

#### Changes proposed in this Pull Request:
This PR moves the following products to the Search package:

* Classic Search
* Instant Search along with its build process
* Customberg along with its build process

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Testing auto-configuration & Instant Search:
1. Activate this branch on a brand new site **without** a Jetpack Search subscription.
2. Make sure to build both Instant Search and Customberg via `pnpm build` within the Search package directory.
3. Purchase/add a Jetpack Search subscription to the site.
4. Ensure that the search module and instant search are both automatically enabled.
5. Ensure that a Jetpack Search with preconfigured filters is added to the site sidebar
6. Perform a site search. Ensure that the overlay uses the Expanded result format and that the sidebar renders the aforementioned search widget from (5).

Testing Customberg (search configuration interface)

7. Navigate to Customberg (`/wp-admin/admin.php?page=jetpack-search-configure`).
8. Ensure Customberg reads and saves instant search configuration settings as expected.

Testing Classic Search 

9. Disable Instant Search using the Jetpack Search dashboard.
10. Add a Jetpack Search widget with configured filters to any available site sidebar.
11. Ensure that the filters from (8) render as expected.
12. Ensure that site search functionality works as expected.

Also, ensure that `pnpm test` passes within the Search package.
